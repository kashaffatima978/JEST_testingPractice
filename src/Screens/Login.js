import React, { useState } from "react"
import {View,Text,StyleSheet,Image, TextInput, Button} from "react-native"



const Login = ()=>{
    const [users, setUser] = ([{name: '', email:''}])
    const[ name, setName]=useState('');
    const[ email, setEmail]=useState('');

    function settingData(name, email){
        setUser([...users, {name: name , email: email}]);
        return true;
    }
    var getData =()=>{
        return{name:name,email: email};
    }

return(
    <View>
        <Text>Name:</Text>
        <TextInput value="name" onChangeText={setName}></TextInput>
        <Text>Email:</Text>
        <TextInput value="email" onChangeText={setEmail}></TextInput>
        <Button onPress={()=>{console.log('hello')}} title={'Login'}/>
    </View>
)
    
    
}

export default Login;