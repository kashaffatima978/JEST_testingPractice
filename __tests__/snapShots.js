import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../src/Screens/Home';
import Login from '../src/Screens/Login';

test('Home', () => {
  const HomeSnap = renderer.create(<Home />).toJSON();

  expect(HomeSnap).toMatchSnapshot();
});

test('login', () => {
  const loginSnap = renderer.create(<Login />).toJSON();

  expect(loginSnap).toMatchInlineSnapshot(`
    <View>
      <Text>
        Name:
      </Text>
      <TextInput
        onChangeText={[Function]}
        value="name"
      />
      <Text>
        Email:
      </Text>
      <TextInput
        onChangeText={[Function]}
        value="email"
      />
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  `);
});
