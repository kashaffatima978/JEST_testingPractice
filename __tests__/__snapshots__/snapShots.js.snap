// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 1`] = `
<View>
  <Text>
    Home Space
  </Text>
  <Text>
    What is Lorem Ipsum?
  </Text>
</View>
`;
